File: internal/model/purchase.go
package model

import "time"

type PurchaseOrder struct {
	ID           string              `gorm:"primaryKey"`
	SupplierID   string              `gorm:"size:191"`
	Supplier     Supplier            `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	WarehouseID  string              `gorm:"size:191"`
	Warehouse    Warehouse           `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	UserID       string              `gorm:"size:191"`
	User         User                `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	IsValidated  bool                `gorm:"default:false"`
	Items        []PurchaseOrderItem `gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	AccountID    string              `gorm:"size:191"`
	Account      Account             `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	OtherCost    float64
	TotalCost    float64
	ActalPayment float64
	Debt         float64
	CreatedAt    time.Time
	Basic        string // 用于退货单
	Kind         string
}

type PurchaseOrderItem struct {
	PurchaseOrderID string  `gorm:"primaryKey;size:191"`
	ProductID       string  `gorm:"primaryKey;size:191"`
	Product         Product `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	Quantity        int
	Price           float64
}

File: web/src/@types/purchase.ts
type PurchaseOrder = {
  id: string;
  warehouse_id: string;
  supplier_id: string;
  user_id: string;
  account_id: string;
  other_cost: number;
  total_cost: number;
  actal_payment: number;
  debt: number;
  created_at: string;
  is_validated: boolean;
  kind: string;
  basic?: string;
  items: Item[];
};

type Item = {
  product_id: string;
  quantity: number;
  price: number;
};
